packages/dds/tree/src/feature-libraries/treeTextCursor.ts

```diff
@@ -89,65 +76,47 @@ export class TextCursor implements ITreeCursor {
     }
 
     get keys(): Iterable<FieldKey> {
-        return Object.getOwnPropertyNames(this.getFields()) as Iterable<FieldKey>;
+        return Object.getOwnPropertyNames(getGenericTreeFieldMap(this.getNode(), false)) as Iterable<FieldKey>;
     }
 
-    down(key: FieldKey, index: number): TreeNavigationResult {
-        const siblings = this.getField(key);
+    down(key: FieldKey, index: number): SynchronousNavigationResult {
+        const siblings = getGenericTreeField(this.getNode(), key, false);
         const child = siblings[index];
         if (child !== undefined) {
-            this.nodeStack.push(child);
-            this.indexStack.push(index);
-            this.keyStack.push(key);
```

Comment:
Possibly, it's worthwhile to continue maintaining the keyStack so that we can construct the path to the current cursor position in the future (e.g., for creating an anchor?)

...just a thought.  No objection to removing it and re-adding later if needed.